src/atldld/sync.py

```diff
@@ -519,6 +520,174 @@ def corners_sagittal(r, dataset_id):
+
+    return df  # type: ignore
+
+
+def download_dataset_parallel(
+    dataset_id: int,
+    ds_r: int = 25,
+    detection_xy: Tuple[Union[int, float], Union[int, float]] = (0, 0),
+    include_expression: bool = False,
+    ds_i: int = 0,
+) -> Generator[
+    Union[
+        Tuple[int, float, np.ndarray, DisplacementField],
+        Tuple[int, float, np.ndarray, DisplacementField, np.ndarray],
+    ],
+    None,
+    None,
+]:
+    """Download entire dataset.
+
+    This function performs the following steps:
+
+    1. Get metadata for the entire dataset (e.g. `matrix_3d`)
+    2. Get metadata for all images inside of the dataset (e.g. `matrix_2d`)
+    3. For each image in the dataset do the following
+
+        a. Query the API to get the `p, i, r` coordinates of the `detection_xy`.
+        b. One of the `p, i, r` will become the `slice_coordinate`. For
+           coronal datasets it is the `p` and for sagittal ones it is the `r`.
+           In other words we assume that the slice is parallel to
+           one of the axes.
+        c. Use `get_transform_parallel` to get a full mapping between the
+           reference space and the image.
+        d. Download the image (+ potentially the expression image)
+        e. Yield result (order derived from section numbers - highest first)
+
+    Parameters
+    ----------
+    dataset_id : int
+        Id of the section dataset. Used to determine the 3D matrix.
+    ds_r : int, optional
+        Downsampling factor of the reference
+        space. If set to 1 no downsampling takes place. The reference
+        space shape will be divided by `ds_r`.
+    detection_xy : tuple
+        Represents the x and y coordinate in the image that will be
+        used for determining the slice number in the reference space.
+        `p` for coronal slices, `r` for sagittal slices.
+    include_expression : bool
+        If True then the generator returns 5 objects
+        where the last one is the expression image.
+    ds_i : int
+        The downloaded image will have both the height and the width
+        downsampled by `2 ** ds_i`.
+
+    Returns
+    -------
+    res_dict : generator
+        Generator yielding consecutive four tuples of
+        (image_id, constant_ref_coordinate, img, df).
+        The `constant_ref_coordinate` is the dimension in the given axis in microns.
+        The `img` is the raw gene expression image with dtype `uint8`.
+        The `df` is the displacement field of shape (8000 // `ds_r`, 11400 // `ds_r`).
+        Note that the sorting. If `include_expression=True` then last returned image
+        is the processed expression image.
+        That is the generator yield (image_id, p, img, df, img_expr).
+    """
+    metadata_2d = get_2d_bulk(
+        dataset_id,
+        ref2inp=True,
+        add_last=False,
+    )
+    matrix_3d = get_3d(
+        dataset_id,
+        ref2inp=True,
+        add_last=False,
+        return_meta=False,
+    )
```

Comment:
My PyCharm is suggesting that one should remove `add_last=False` and `return_meta=False` because the values are equal the default values.

Reply:
Hmmm...I guess this is not something `flake8` would catch, right? Anyway, fixed in 47fdecd6a186c272be635cb55854e139d53db7ed